### PR TITLE
Use precompiled extensions only

### PR DESCRIPTION
**Pull request by soltanmm**
cc @jtattermusch @hsaliak @ejona86 

I'm going to run this on Jenkins by pointing it at the relevant branch. If things go well then right after that the PR can go in.

### CODE REVIEW

**Comment by soltanmm** [comment]
This is now staged on TestPyPI.

File: tools/run_tests/build_artifact_python.sh [hunk 1]
```diff
@@ -46,7 +46,7 @@ fi
 GRPC_PYTHON_USE_CUSTOM_BDIST=0  \
 GRPC_PYTHON_BUILD_WITH_CYTHON=1 \
 ${SETARCH_CMD} python setup.py  \
-    bdist_wheel_grpc_custom
+    build_tagged_ext
 
 GRPC_PYTHON_USE_CUSTOM_BDIST=0  \
 GRPC_PYTHON_BUILD_WITH_CYTHON=1 \
```
Comment (jtattermusch) [inline tools/run_tests/build_artifact_python.sh]: bdist_wheel should only be built on mac, but for now it's fine (let's fix this in a followup PR)

**Comment by jtattermusch** [comment]
The approach looks fine to me (considering where we are right now). I am happy with the results of the linux distribtests. If we get positive feedback from a friction log, we can merge.

**Comment by jtattermusch** [comment]
So basically, LGTM but let's upload to testpypi and test it manually.

**Comment by hsaliak** [comment]
I can confirm that this worked manually.
I also retried after removing python-dev and python-all-dev to ensure that I lack dependencies to compile the code.

**Comment by soltanmm** [comment]
BEFORE I FORGET (and this needs to be written down somewhere):

This method works for `pip install`, but directly invoking `python setup.py install` will create an unreachable `cygrpc.so` on *some* systems. I believe this can be papered over, but that's for a day when we're not all `X_X`.

**Pull request merged**